### DISCUSSION TITLE
UseQuery useErrorBoundary config field inverted?

### DISCUSSION THREAD

**Discussion by sean-esper** [TanStack/query#1198] [Q&A]
In the useQuery config type file I find:
```
/**
   * Whether errors should be thrown instead of setting the `error` property.
   * Defaults to `false`.
   */
  useErrorBoundary?: boolean;
```

This actually seems inverted. The default behavior (false) seems to be for react-query to add a boundary and prevent the error from bubbling up. I have a custom errorBoundary on my site and therefore to get the error to bubble up I have to set this field to `true` which seems like (according to the field name) that it would add an error boundary rather than remove it.

Am I misunderstanding something or should I start drafting up a PR?

**Comment by sergiodxa** [comment]
React Query by default doesn't throw on an error, instead you get the error as a return of the useQuery
```ts
const { data, error } = useQuery(..queryArgs)
```
When you enable `useErrorBoundary` the hook will throw an error instead of giving you this error value, this way you can use an error boundary to catch the error.

**Reply by sean-esper** [reply]
Yea I understand what its doing. I guess what I'm saying is Id expect that since react-query doesn't throw by default, that it is using an errorboundary by default, yet the field is called useErrorBoundary with a default value of false.

I'd expect this field to be called throwOnError (like it is in useMutation) with a default of false (which would mean for react-query to use an errorBoundary)

**Reply by sergiodxa** [reply]
Now I get your point, it makes sense to be `throwOnError` which is also consistent with `useMutation`.

Note that hooks can't setup an error boundaries tho, so I think the current name comes from that, since a hook can't setup error boundaries to use them you need to throw.
